src/pod.rs

```diff
@@ -0,0 +1,320 @@
+use self::super::{Error, guarded_transmute_vec_permissive, guarded_transmute_many_permissive, guarded_transmute_many_pedantic, guarded_transmute_vec_pedantic,
+                  guarded_transmute_pedantic, guarded_transmute_vec, guarded_transmute_many, guarded_transmute};
+
+
+/// Type that can be non-`unsafe`ly transmuted into
+///
+/// In most cases this is a [*POD class*](http://eel.is/c++draft/class#10)
+/// or a [*trivially copyable class*](http://eel.is/c++draft/class#6).
+///
+/// Marker trait for `guarded_transmute_pod_*()` functions.
+///
+/// *Nota bene*: `bool`s aren't *actually* non-`unsafe` to transmute, because they're restricted to being `0` or `1`,
+/// which means it's UB to transmute an arbitrary byte to into a `bool`.
+///
+/// *Nota bene*: if you transmute into a floating-point type you will have a chance to create signaling NaNs
+/// (judgement's still out as to whether they're legal).
+pub trait PodTransmutable {}
+
+impl PodTransmutable for u8 {}
+impl PodTransmutable for i8 {}
+impl PodTransmutable for u16 {}
+impl PodTransmutable for i16 {}
+impl PodTransmutable for u32 {}
+impl PodTransmutable for i32 {}
+impl PodTransmutable for u64 {}
+impl PodTransmutable for i64 {}
+impl PodTransmutable for char {}
+#[cfg(stage0)]
```

Comment:
Is this `cfg()` argument right, btw?

Reply:
Hmm, `byteorder` seems to be using `#[cfg(feature = "i128")]`. IMO we should do the same.

Reply:
Ah, cool. I stole this from `libcore` so I wasn't sure.

Reply:
I mean, it has its own conditional feature for this ([code](https://github.com/BurntSushi/byteorder/blob/master/src/lib.rs#L41)). The real Rust compiler feature for these types is `feature(i128_type)`.

Reply:
Does that mean that `#[cfg(i128_type)]` is correct, or?